File: app/src/main/java/com/haya/tablayout_viewpager_java/CustomViewPager.java
//package com.haya.tablayout_viewpager_java;
//
//import android.view.MotionEvent;
//
//import androidx.viewpager.widget.ViewPager;
//
//public class CustomViewPager extends ViewPager {
//
//    @Override
//    public boolean onInterceptTouchEvent(MotionEvent arg0) {
//        // Never allow swiping to switch between pages
//        if(swipeLock){
//            return super.onInterceptTouchEvent(arg0);
//        }
//        else{
//            return false;
//        }
//    }
//
//    @Override
//    public boolean onTouchEvent(MotionEvent event) {
//        // Never allow swiping to switch between pages
//        if(swipeLock){
//            return super.onTouchEvent(event);
//        }
//        else{
//            return false;
//        }
//    }
//}

File: app/src/main/java/com/haya/tablayout_viewpager_java/CustomViewPager.kt
package com.haya.tablayout_viewpager_java

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.viewpager.widget.ViewPager

class CustomViewPager(context: Context, attrs: AttributeSet?) : ViewPager(context, attrs) {

    private var isEnable = true

    override fun onTouchEvent(ev: MotionEvent?): Boolean = when (isEnable) {
        true -> super.onTouchEvent(ev)
        else -> false
    }

    override fun onInterceptTouchEvent(ev: MotionEvent?): Boolean = when (isEnable) {
        true -> super.onInterceptTouchEvent(ev)
        else -> false
    }

    fun setPagingEnabled(isEnable: Boolean) {
        this.isEnable = isEnable
    }
}
